src/main/webui/src/targets/TargetTable.tsx

```diff
@@ -270,4 +223,35 @@ export function TargetTableRow(props: TargetProps): ReactElement {
         </Table.Tr>);
 }
 
-export default TargetPanel
\ No newline at end of file
+/**
+ * generates a table for an entire set of targets.
+ * @param {TargetTableProps} props the data required to generate the table.
+ * @return {React.ReactElement} the html that contains the table.
+ * @constructor
+ */
+export function TargetTable(props: TargetTableProps): ReactElement {
+    return (
+        <Table>
+            {TargetTableHeader(props)}
+            <Table.Tbody>
+                {props.isLoading ? (
+                        <Table.Tr colSpan={5} key={randomId()}>
+                            <Table.Td>Loading...</Table.Td>
+                        </Table.Tr>) :
+                    props.data?.map((item) => {
+                        return (
+                            <TargetTableRow
+                                proposalCode={Number(props.selectedProposalCode)}
+                                dbid={item.dbid!}
+                                showButtons={props.showButtons}
+                                key={randomId()}
```

Comment:
```suggestion
                                key={String(item.dbid!}
```
otherwise it's generating a new random key every render which defeats the purpose of a key

Reply:
fixed. i tried to just have key and access that, but it seems your not even allowed to access the key variable. so its really a nasty extra bit of data to store. but done